go-controller/pkg/ovn/base_network_controller_pods.go

```diff
@@ -574,12 +574,14 @@ func (bnc *BaseNetworkController) addLogicalPortToNetwork(pod *kapi.Pod, nadName
 		// IP/MAC from the annotation.
 		lsp.DynamicAddresses = nil
 
-		// ensure we have reserved the IPs in the annotation
-		if err = bnc.lsManager.AllocateIPs(switchName, podIfAddrs); err != nil && err != ipallocator.ErrAllocated {
-			return nil, nil, nil, false, fmt.Errorf("unable to ensure IPs allocated for already annotated pod: %s, IPs: %s, error: %v",
-				podDesc, util.JoinIPNetIPs(podIfAddrs, " "), err)
-		} else {
-			needsIP = false
+		if bnc.doesNetworkRequireIPAM() {
```

Comment:
Should we fail if `bnc.doesNetworkRequireIPAM()==false` and `len(podIfAddrs) > 0`

The annotated IPs are specific to OVN IPAM and should not be set?

Reply:
Yeah, it should never happen from the perspective that we should not set the annotation ourselves.

But nothing prevents a user from doing so. We have hard weird cases where users were creating pods from previous existing pods and who know what they are exactly doing.

What would be the outcome of this scenario if we didn't do this check? We would be creating the LSP with those IP addresses which is something we wouldn't want and then who knows what happens. 

So probably safer to do the check?